### PR TITLE
Upgrade to doctrine/coding-standard 10

### PR DESCRIPTION
**Pull request by greg0ire**
This reverts commit 2393612223c555b113b287ac2ac49b2072bd6374.

### CODE REVIEW

File: src/Platforms/AbstractMySQLPlatform.php [hunk 1]
```diff
@@ -439,8 +439,11 @@ public function getCreateTablesSQL(array $tables): array
                 Deprecation::trigger(
                     'doctrine/dbal',
                     'https://github.com/doctrine/dbal/pull/5414',
-                    'Relying on the DBAL not generating DDL for foreign keys on MySQL engines'
-                        . ' other than InnoDB is deprecated. Define foreign key constraints only if they are necessary.',
+                    <<<'DEPRECATION'
+                    Relying on the DBAL not generating DDL for foreign keys on
+                    MySQL engines other than InnoDB is deprecated.
+                    Define foreign key constraints only if they are necessary.
+                    DEPRECATION,
```
Comment (greg0ire) [inline src/Platforms/AbstractMySQLPlatform.php]: If having the message on one line is important, I can use `str_replace` to remove newlines. Let me know.

Comment (morozov) [reply src/Platforms/AbstractMySQLPlatform.php]: As I understand, the problem is that the newly added column makes the line longer than 120 characters. In this case, I'd break it down in two using concatenation. IMO, `str_replace()` would be an even worse hack than concatenation. And yes, I'd prefer to have the message on one line leaving the formatting to the presentation layer.

File: tests/Functional/Schema/SchemaManagerFunctionalTestCase.php [hunk 2]
```diff
@@ -433,9 +433,9 @@ public function testDispatchEventWhenDatabasePlatformIsExplicitlyPassed(): void
     public function testDiffListTableColumns(callable $comparatorFactory): void
     {
         if ($this->connection->getDatabasePlatform() instanceof OraclePlatform) {
-            self::markTestSkipped(
-                'Does not work with Oracle, since it cannot detect DateTime, Date and Time differences (at the moment).',
-            );
+            self::markTestSkipped(<<<'MSG'
```
Comment (morozov) [inline tests/Functional/Schema/SchemaManagerFunctionalTestCase.php]: Should we just use concatenation instead?

**Review by morozov — COMMENTED** [review]
As for the long strings, in some languages using concatenation to declare a long string in order to satisfy the coding standard is frowned upon since it makes the string harder to search in the code.

I'm curious if the standard could make an exception for such string (e.g. it's the only token on the line excluding whitespace and comma).

**Comment by greg0ire** [comment]
For the record, I think we should use linebreaks in messages a bit more, one reason being it would make us feel a bit more free to write longer messages with more details. I don't know how the cs rule about line breaks is implemented, but I'd be surprised if it involved a parser.

Since the source code and the messages are both read by humans, it would make sense to make the messages legible by having linebreaks at some point. Using unindented nowdoc might allow having the same line length limit, except in situations where the message is used in an HTML without the linebreaks being transformed to `<br/>`. I'd be curious to know what presentation layers out there currently treat linebreaks.

**Review by morozov — APPROVED** [review]

**Pull request merged**